backend/src/schema/resolvers/notifications.js

```diff
@@ -42,20 +42,28 @@ export default {
         default:
           orderByClause = ''
       }
-      const offset = args.offset && typeof args.offset === 'number' ? `SKIP ${args.offset}` : ''
-      const limit = args.first && typeof args.first === 'number' ? `LIMIT ${args.first}` : ''
 
-      const readTxResultPromise = session.readTransaction(async transaction => {
-        const notificationsTransactionResponse = await transaction.run(
-          ` 
-          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
+      const returnClause = `
           ${whereClause}
           WITH user, notification, resource,
           [(resource)<-[:WROTE]-(author:User) | author {.*}] as authors,
           [(resource)-[:COMMENTS]->(post:Post)<-[:WROTE]-(author:User) | post{.*, author: properties(author)} ] as posts
           WITH resource, user, notification, authors, posts,
           resource {.*, __typename: labels(resource)[0], author: authors[0], post: posts[0]} as finalResource
           RETURN notification {.*, from: finalResource, to: properties(user)}
+      `
+
+      const offset = args.offset && typeof args.offset === 'number' ? `SKIP ${args.offset}` : ''
+      const limit = args.first && typeof args.first === 'number' ? `LIMIT ${args.first}` : ''
+
+      const readTxResultPromise = session.readTransaction(async transaction => {
+        const notificationsTransactionResponse = await transaction.run(
+          `
+          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
+          ${returnClause}
+          UNION
+          MATCH (user)-[:WROTE]->(resource)<-[:BELONGS_TO]-(report:Report)-[notification:NOTIFIED]->(user:User {id:$id})
+          ${returnClause}
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-12T17:07:48Z" title="Wednesday, February 12th 2020, 6:07:48 pm +01:00">Feb 12, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -42,20 +42,28 @@ export default {
         default:
           orderByClause = ''
       }
-      const offset = args.offset && typeof args.offset === 'number' ? `SKIP ${args.offset}` : ''
-      const limit = args.first && typeof args.first === 'number' ? `LIMIT ${args.first}` : ''
 
-      const readTxResultPromise = session.readTransaction(async transaction => {
-        const notificationsTransactionResponse = await transaction.run(
-          ` 
-          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
+      const returnClause = `
           ${whereClause}
           WITH user, notification, resource,
           [(resource)<-[:WROTE]-(author:User) | author {.*}] as authors,
           [(resource)-[:COMMENTS]->(post:Post)<-[:WROTE]-(author:User) | post{.*, author: properties(author)} ] as posts
           WITH resource, user, notification, authors, posts,
           resource {.*, __typename: labels(resource)[0], author: authors[0], post: posts[0]} as finalResource
           RETURN notification {.*, from: finalResource, to: properties(user)}
+      `
+
+      const offset = args.offset && typeof args.offset === 'number' ? `SKIP ${args.offset}` : ''
+      const limit = args.first && typeof args.first === 'number' ? `LIMIT ${args.first}` : ''
+
+      const readTxResultPromise = session.readTransaction(async transaction => {
+        const notificationsTransactionResponse = await transaction.run(
+          `
+          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
+          ${returnClause}
+          UNION
+          MATCH (user)-[:WROTE]->(resource)<-[:BELONGS_TO]-(report:Report)-[notification:NOTIFIED]->(user:User {id:$id})
+          ${returnClause}
```


I have not understand where the reason categories and the explanatory text stored in the future in the report will come into the response here.
You making your life easy by not caring for issue #2885 `Moderator Decision Shall Have Reason And Information`  which originally would have been implemented before this, like I told you in our conversations and send you in a private message @roschaefer …